node/src/gql/types/user-access-token.ts

```diff
@@ -23,6 +25,75 @@ export function accessTokenDuration(): number {
     : 60 * 60 * 24 * 90;
 }
 
+export async function getRefreshedToken(token: string) {
+  const refreshToken = await getRefreshToken(token);
+  const { user } = refreshToken;
+
+  // replace old refresh token with a new one and save
+  const newRefreshToken = await generateRefreshToken(user);
+  refreshToken.revoked = new Date(Date.now());
+  refreshToken.replacedByToken = newRefreshToken.token;
+  await refreshToken.save();
+  await newRefreshToken.save();
+
+  // generate new jwt
+  const jwtToken = generateJwtToken(user);
+  return { jwtToken, newRefreshToken, user };
+  // return basic details and tokens
+  // return {
+  //     ...basicDetails(user),
+  //     jwtToken,
+  //     refreshToken: newRefreshToken.token
+  // };
+}
+
+async function getRefreshToken(token: string) {
+  const refreshToken = await RefreshTokenSchema.findOne({ token }).populate(
+    'user'
+  );
+  if (!refreshToken || !refreshToken.isActive) throw 'invalid token';
+  return refreshToken;
+}
+
+export function setTokenCookie(res: Response, token: string) {
+  // create http only cookie with refresh token that expires in 7 days
+  const cookieOptions = {
+    httpOnly: true,
+    expires: new Date(Date.now() + 7 * 24 * 60 * 60 * 1000),
+  };
+  res.cookie('refreshToken', token, cookieOptions);
+}
+
+function randomTokenString() {
+  return randomBytes(40).toString('hex');
+}
+
+export function generateRefreshToken(user: User) {
+  // create a refresh token that expires in 7 days
+  return new RefreshTokenSchema({
+    user: user.id,
+    token: randomTokenString(),
+    expires: new Date(Date.now() + 7 * 24 * 60 * 60 * 1000),
```

Comment:
7 days? I thought the asana task said 3 months

Reply:
...also should be configurable by an env var and then 3 months would be default if env var not set (or invalid)